pppd: Cap MTU to the user configured value

This patchs caps the calculated MTU value in lcp.c to the user specified "mru"
option value. Without this patch pppd would advertise a different MTU value
compared to what is set on the local interface in some cases.

Signed-off-by: Jo-Philipp Wich <jo@mein.io>

--- a/pppd/lcp.c
+++ b/pppd/lcp.c
<<<<<<< HEAD
@@ -1888,12 +1888,12 @@ lcp_up(fsm *f)
=======
@@ -1916,12 +1916,12 @@ lcp_up(f)
>>>>>>> 712839d4c6 (Removed unwanted submodules from index)
      * the interface MTU is set to the lowest of that, the
      * MTU we want to use, and our link MRU.
      */
-    mtu = ho->neg_mru? ho->mru: PPP_MRU;
+    mtu = MIN(ho->neg_mru? ho->mru: PPP_MRU, ao->mru);
     mru = go->neg_mru? MAX(wo->mru, go->mru): PPP_MRU;
<<<<<<< HEAD
 #ifdef PPP_WITH_MULTILINK
     if (!(multilink && go->neg_mrru && ho->neg_mrru))
 #endif /* PPP_WITH_MULTILINK */
-	ppp_set_mtu(f->unit, MIN(MIN(mtu, mru), ao->mru));
+	ppp_set_mtu(f->unit, MIN(mtu, mru));
=======
 #ifdef HAVE_MULTILINK
     if (!(multilink && go->neg_mrru && ho->neg_mrru))
 #endif /* HAVE_MULTILINK */
-	netif_set_mtu(f->unit, MIN(MIN(mtu, mru), ao->mru));
+	netif_set_mtu(f->unit, MIN(mtu, mru));
>>>>>>> 712839d4c6 (Removed unwanted submodules from index)
     ppp_send_config(f->unit, mtu,
 		    (ho->neg_asyncmap? ho->asyncmap: 0xffffffff),
 		    ho->neg_pcompression, ho->neg_accompression);
